File: design-patterns/JavaDesignPatterns/src/com/bigboxcode/designpattern/templatemethod/transport/Car.java
// Car.java

package com.bigboxcode.designpattern.templatemethod.transport;

public class Car extends Transport {
    @Override
    void createBody() {
        System.out.println("Creating Car Body");
    }

    @Override
    void addEngine() {
        System.out.println("Adding Engine to Car");
    }

    @Override
    void addWheel() {
        System.out.println("Adding 4 Wheels to Car");
    }

    @Override
    void addWing() {
        // Not required for Car
    }
}

File: design-patterns/ts/template-method/transport/car.ts
// car.ts

import Transport from "./transport";

class Car extends Transport {
    createBody(): void {
        console.log("Creating Car Body");
    }

    addEngine(): void {
        console.log("Adding Engine to Car");
    }

    addWheel(): void {
        console.log("Adding 4 Wheels to Car");
    }

    addWing(): void {
        // Not required for Car
    }
}

export default Car;